infra/agentInterface.go

```diff
@@ -15,4 +15,6 @@ type IAgent[T any] interface {
 	NotifyAgentFinishedMessaging()
 	// allows for synchronous messaging to be run
 	RunSynchronousMessaging()
+	GetListeningSpinnerFlag() bool
+	SetListeningSpinner(bool)
```

Comment:
This variable is a member of the baseagent struct and is just used as an exit condition and is triggered by the agent so the user can control exactly when the listening spinner exits. Setting it to true will cause the listening spinner to exit on the next iteration of the for loop. Before exiting the spinner it will automatically set the variable back to false so the spinner doesn't automatically exit when it is started again on the next turn.